src/shims/windows/handle.rs

```diff
@@ -126,21 +127,32 @@ impl Handle {
         Scalar::from_target_isize(signed_handle.into(), cx)
     }
 
-    pub fn from_scalar<'tcx>(
+    /// fail if thread id is invalid
```

Comment:
```suggestion
    /// Convert a scalar into a structured `Handle`.
    /// Structurally invalid handles return `Ok(None)`.
    /// If the handle is structurally valid but semantically invalid, e.g. a for non-existent thread
    /// ID, returns `Err`.
```
This makes me wonder if that's really the semantics we want. Wouldn't `Result<Self, ...>` make more sense?

Reply:
> This makes me wonder if that's really the semantics we want. Wouldn't `Result<Self, ...>` make more sense?

Make sense, this seems a bit repeating, how about `Result<Self, HandleError>` with:
```rust
enum HandleError{
  ThreadNotFound(ThreadNotFound),
  InvalidHandle
}
```